src/blob/index/core.rs

```diff
@@ -296,27 +290,41 @@ where
     }
 
     async fn get_all(&self, key: &K) -> Result<Option<Vec<RecordHeader>>> {
-        match &self.inner {
+        let res = match &self.inner {
             State::InMemory(headers) => Ok(headers.get(key).cloned()),
```

Comment:
In memory all records stored in order of addition: the latest record is at the end. So you need to reverse the order here

Reply:
Before calling `reverse` please check, that the length of the vector is `> 1`